docs/types.rst

```diff
@@ -556,37 +558,73 @@ Type Conversions
 
 All type conversions in Vyper must be made explicitly using the built-in ``convert(a: atype, btype)`` function. Currently, the following type conversions are supported:
 
-================ ================== ============================== =================================
-In (``atype``)   Out (``btype``)    Allowable Values               Additional Notes
-================ ================== ============================== =================================
-``bool``         ``decimal``        All                            ``0.0`` or ``1.0``
-``bool``         ``int128``         All                            ``0`` or ``1``
-``bool``         ``uint256``        All                            ``0`` or ``1``
-``bool``         ``bytes32``        All                            ``0x00`` or ``0x01``
-``bool``         ``Bytes``          All
-``decimal``      ``bool``           All                            Returns ``a != 0.0``
-``decimal``      ``int128``         All                            Value is truncated
-``decimal``      ``uint256``        ``a >= 0.0``                   Value is truncated
+================ ================== ==================================== ==================================
+In (``atype``)   Out (``btype``)    Allowable Values                     Additional Notes
+================ ================== ==================================== ==================================
+``address``      ``bool``           All                                  Returns ``a != ZERO_ADDRESS``
+``address``      ``decimal``        All                                  | Value is truncated to the
+                                                                         | least significant 128 bits
+``address``      ``int128``         All                                  | Value is truncated to the
+                                                                         | least significant 128 bits
```

Comment:
Is the value truncated? Or will it revert if exceeded?

Reply:
The value will be truncated. For example, `0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF` evaluates to `-1` for `int128` and `Fixed(-1)` for `decimal`.

Reply:
What about `0x80000000000000000000FFFFFFFFFFFFFFFFFFFF`?

Reply:
It truncates as well - `1208925819614629174706175` and `Fixed('1208925819614629174706175')` for `int128` and `decimal` respectively. Is there an intuition to this test case? I can try exploring in that direction.

Reply:
Yes, what happens to a value outside of the `int128` range when converted?

Reply:
From my understanding, in this case of converting from `address` to `int128` and `decimal`, the input address is sign-extended from 16 bytes to 32 bytes, so the most significant 4 bytes of the input address are dropped. Consequently, the remaining 16 bytes will always be a value that can be represented in `int128`. Perhaps 'truncated' is not the right way to phrase it?

Reply:
> Perhaps 'truncated' is not the right way to phrase it?

yeah, thinking about this

Reply:
How about these alternatives? I think this would be more aligned with the actual operation.

> Extracts the rightmost sixteen bytes only
> Leftmost four bytes are discarded

Reply:
sure!